src/adhocracy_core/docs/workflows.rst

```diff
@@ -0,0 +1,119 @@
+Workflows
+==========
+
+Preliminaries
+-------------
+
+For testing, we need to import some stuff and start the Adhocracy testapp::
+
+    >>> from pprint import pprint
+    >>> from adhocracy_core.resources.pool import IBasicPool
+    >>> from adhocracy_core.resources.sample_proposal import IProposal
+    >>> app_god = getfixture('app_god')
+    >>> app_god.base_path = '/adhocracy'
+
+Lets create some content::
+
+    >>> data = {'adhocracy_core.sheets.name.IName': {'name': 'proposals'}}
+    >>> resp = app_god.post_resource('/', IBasicPool, data)
+    >>> data = {'adhocracy_core.sheets.name.IName': {'name': 'proposal_item'}}
+    >>> resp = app_god.post_resource('/proposals', IProposal, data)
+
+
+Workflows
+---------
+
+Workflows are fine state machines Assigned to a Resource.
+States can set the local permissions.
+States can have metadata (title, date,...).
+State transitions can have a callable to excecute arbitrary tasks.
+TODO nice Introduction.
+
+The MetaAPI gives us the states and transitions metadata for each workflow::
+
+    >>> resp_data = app_god.get('/../meta_api').json
+    >>> worklow = resp_data['worklows']['sampleworkflow']
+
+State metadata contains a human readable title ::
+
+    >>> state = workflow['states']['announced']
+    >>> state['title']
+    'Announced'
+
+a description ::
+
+    >>> state['description']
+    'In this phase you may wait.'
+
+a local ACL (see doc:`authorization`) that is set when entering this state, ::
+
+    >>> state['acl']
+    [('Disallow, 'role:reader', 'view')]
+
+a hint for the frontend if displaying this state in listing should be restricted::
+
+    >>> state['display_only_to_roles']
+    ['Manager']
+
+The order these states shoudl be listet is also set, in addition this
+defines the initial workflow state (the first in the list). ::
+
+    >>> workflow['states_order']
+    ['draft', 'announced', 'published']
+
+Transition metadata determes the possible state flow and can provide a callable to
+excecute arbitrary tasks ::
+
+     >>> transition = workflow['transitions']['to_announced']
+     >>> pprint(transition)
+     {
+      'callback': 'adhocracy_core.workflows.samplecallback',
+      'from_state': 'draft',
+      'to_state': 'announced',
+      'permission': 'review',
+      }
```

Comment:
> What does "callable" mean in this context? Is this an API endpoint? If it is an actual python object, why should it be exposed to the frontend?
> 
> Exposing the callable makes sense if the frontend wants to create new workflows in the future.
> Maybe for now its more usefull to display the callable docstring instead. Then the frontend developer knows better whats happening when entering this state.

ok, we remove callable and permission because this is not usefull for the frontend at the moment.
In future we may add the docstring explain what this transition does.